src/hotspot/share/utilities/globalDefinitions.cpp

```diff
@@ -323,7 +323,7 @@ int _type2aelembytes[T_CONFLICT+1] = {
 
 #ifdef ASSERT
 int type2aelembytes(BasicType t, bool allow_address) {
-  assert(allow_address || t != T_ADDRESS, " ");
+  assert((allow_address || t != T_ADDRESS) && t != T_ILLEGAL, "unexpected basic type");
```

Comment:
This is OK, but it only handles one possible bad value.  How about checking that t <= T_CONFLICT, or t < ARRAY_SIZE(__type2aelembytes)?

Reply:
good point! I updated it